### ISSUE TITLE
Incompatible with "withSourcesJar"

### ISSUE THREAD

**Issue by Brummolix** [boothen/gradle-wsimport#23]
The following build.gradle using java withSourcesJar and running on Gradle 8.0 does not work:

```groovy
plugins {
    id "java-library"
    id 'uk.co.boothen.gradle.wsimport' version "0.18"
}

group = 'org.example'
version = '1.0-SNAPSHOT'

repositories {
    mavenCentral()
}

java {
    withSourcesJar()
}

wsimport {
    wsdlSourceRoot = 'src/main/wsdl'
    wsdl('example.wsdl')
}

//uncomment this to make build task work
//tasks.named("sourcesJar") {
//    dependsOn 'wsImport1'
//}

```

If you call the _build_ task, it gives the following error:
```
FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':sourcesJar' (type 'Jar').
  - Gradle detected a problem with the following location: 'C:\wsImport\build\generated\src\wsdl\main'.
    
    Reason: Task ':sourcesJar' uses this output of task ':wsImport1' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':wsImport1' as an input of ':sourcesJar'.
      2. Declare an explicit dependency on ':wsImport1' from ':sourcesJar' using Task#dependsOn.
      3. Declare an explicit dependency on ':wsImport1' from ':sourcesJar' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.0/userguide/validation_problems.html#implicit_dependency for more details about this problem.
```

**Comment by boothen** [comment]
Hi @Brummolix 

Thanks for reporting the issue. A quick question, is the issue specific to Gradle 8.0?

Cheers,
Matt.

**Comment by Brummolix** [comment]
When you run it with Gralde 7.0 to 7.6 it works, but you get a warning:
```
Execution optimizations have been disabled for task ':sourcesJar' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: 'C:\wsImport\build\generated\src\wsdl\main'. Reason: Task ':sourcesJar' uses this output of task ':wsImport1' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on
 what order the tasks are executed. Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#implicit_dependency for more details about this problem.
``` 
Starting with 8.0 it turns into an error.

**Comment by boothen** [comment]
Thanks for the details. I managed to test it out last night.

I've pushed a fix for this, version 0.19 should be available.

Cheers,
Matt.

**Comment by Brummolix** [comment]
Thanks for the quick fix.
